build.go

```diff
@@ -304,6 +306,21 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return err
 	}
 
+	// Default mode: if the TrustBuilder option is not set, trust the suggested builders.
+	if opts.TrustBuilder == nil {
+		opts.TrustBuilder = func() bool {
+			return false
+		}
+		for _, sugBuilder := range builder.SuggestedBuilders {
+			if opts.Builder == sugBuilder.Image {
+				opts.TrustBuilder = func() bool {
+					return true
+				}
+				break
+			}
+		}
+	}
```

Comment:
I don't think this will work as you expect it to. What we're looking to do is delegate the decision to a function that looks at the `builder.SuggestedBuilders` list.

In go (pseudo) code:

```go
// define the default logic as a function
func IsSuggestedBuilderFunc = func(b string) bool {
  for _, suggestedBuilder := range builder.SuggestedBuilders {
      if b == suggestedBuilder {
          return true
      }
  }

  return false
}

...

// set it if it isn't set
if opts.TrustBuilder == nil {
    opts.TrustBuilder = IsSuggestedBuilderFunc
}

// use it
if lifecycleSupportsCreator && opts.TrustBuilder(opts.Builder) {
   ...
}
```

_For reference on the design pattern you can take a look at the [Strategy Pattern](https://springframework.guru/gang-of-four-design-patterns/strategy-pattern/) from Gang of Four._

Reply:
Maybe an easier example of the pattern in Go: https://refactoring.guru/design-patterns/strategy/go/example

Reply:
Ahh gotcha! This is a lot simpler and easier to understand! :)